sense-android-library/src/nl/sense_os/service/phonestate/PhoneActivitySensor.java

```diff
@@ -88,19 +95,61 @@ private void sendData(String screen)
     
     }
 
+    private boolean active;
+    private PowerManager pm;
+    private PeriodicPollAlarmReceiver alarmReceiver;
+
+    /**
+     * Constructor.
+     * 
+     * @param context
+     * @see #getInstance(Context)
+     */
     protected PhoneActivitySensor(Context context) {
+        super();
         this.context = context;
+        pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+        alarmReceiver = new PeriodicPollAlarmReceiver(this);
+    }
+
+    @Override
+    public void doSample() {
+        sendData(pm.isScreenOn()?"on":"off");
```

Comment:
@orhan89 while you're add it, could you do a API version check and use isInteractive() if available?http://developer.android.com/reference/android/os/PowerManager.html#isInteractive()